Code/GraphMol/SmilesParse/test.cpp

```diff
@@ -3178,7 +3178,7 @@ void testFragmentSmiles() {
     std::string csmiles =
         MolFragmentToSmiles(*m, atomsToUse, 0, 0, &bondLabels);
     std::cerr << csmiles << std::endl;
-    TEST_ASSERT(csmiles == "CaC(aC)bC");
+    TEST_ASSERT(csmiles == "CaC(bC)aC" || csmiles == "CbC(ac)ac");
```

Comment:
I admit to be confused by the || here.  Why are there two results?

Reply:
the atom ranking doesn't take the user-provided bond symbols into account and the atoms are otherwise (correctly) identical as far as the canonicalization algorithm is concerned. So you can't be sure which one you'll get.